pkg/cri/server/image_pull.go

```diff
@@ -318,7 +318,7 @@ func (c *criService) getTLSConfig(registryTLSConfig criconfig.TLSConfig) (*tls.C
 		if len(cert.Certificate) != 0 {
 			tlsConfig.Certificates = []tls.Certificate{cert}
 		}
-		tlsConfig.BuildNameToCertificate() // nolint:staticcheck
+		tlsConfig.BuildNameToCertificate() //nolint:staticcheck
```

Comment:
This `//nolint`  was added in https://github.com/containerd/cri/commit/f5c7ac92724405806eb4e330ecab8f4350601089 (https://github.com/containerd/cri/pull/1517)
to suppress warnings about the `NameToCertificate` function being deprecated:

    // Deprecated: NameToCertificate only allows associating a single certificate
    // with a given name. Leave that field nil to let the library select the first
    // compatible chain from Certificates.

Looking at that, I see it was deprecated in Go 1.14 through https://github.com/golang/go/commit/eb93c684d40de4924fc0664d7d9e98a84d5a100b (https://go-review.googlesource.com/c/go/+/205059), which describes:

    crypto/tls: select only compatible chains from Certificates
    
    Now that we have a full implementation of the logic to check certificate
    compatibility, we can let applications just list multiple chains in
    Certificates (for example, an RSA and an ECDSA one) and choose the most
    appropriate automatically.
    
    NameToCertificate only maps each name to one chain, so simply deprecate
    it, and while at it simplify its implementation by not stripping
    trailing dots from the SNI (which is specified not to have any, see RFC
    6066, Section 3) and by not supporting multi-level wildcards, which are
    not a thing in the WebPKI (and in crypto/x509).

I think we should at least have a comment describing why we're ignoring this, but preferably review whether we should still use it. I'm kinda horrible on TLS stuff, so perhaps someone else would be able to assist on this (I'm happy to open a pull request to make the actual changes).